manifests/init.pp

```diff
@@ -1,153 +1,107 @@
-      $autofs_auto_master_default = '/etc/auto.master'
-      $autofs_sysconfig_template  = 'autofs/autofs_linux.erb'
-    }
-    'Solaris': {
-      $autofs_package_default     = 'SUNWatfsr'
-      $autofs_sysconfig_default   = '/etc/default/autofs'
-      $autofs_auto_master_default = '/etc/auto_master'
-      $autofs_sysconfig_template  = 'autofs/autofs_solaris.erb'
-
-    }
-    default: {
-      fail("Operating system family ${::osfamily} is not supported")
-    }
-  }
-
   # variable preparations
-  $autofs_package_real     = pick($autofs_package, $autofs_package_default)
-  $autofs_sysconfig_real   = pick($autofs_sysconfig, $autofs_sysconfig_default)
-  $autofs_auto_master_real = pick($autofs_auto_master, $autofs_auto_master_default)
-
-  case type3x($timeout) {
-    'integer': { $timeout_int = $timeout }
-    'string':  { $timeout_int = $timeout + 0}
-    default:   { fail('autofs::timeout is not an integer.') }
-  }
-
-  case type3x($negative_timeout) {
-    'integer': { $negative_timeout_int = $negative_timeout }
-    'string':  { $negative_timeout_int = $negative_timeout + 0}
-    default:   { fail('autofs::negative_timeout is not an integer.') }
-  }
-
-  case type3x($mount_wait) {
-    'integer': { $mount_wait_int = $mount_wait }
-    'string':  { $mount_wait_int = $mount_wait + 0}
-    default:   { fail('autofs::mount_wait is not an integer.') }
-  }
-
-  case type3x($umount_wait) {
-    'integer': { $umount_wait_int = $umount_wait }
-    'string':  { $umount_wait_int = $umount_wait + 0}
-    default:   { fail('autofs::umount_wait is not an integer.') }
-  }
-
-  case type3x($mount_nfs_default_protocol) {
-    'integer': { $mount_nfs_default_protocol_int = $mount_nfs_default_protocol }
-    'string':  { $mount_nfs_default_protocol_int = $mount_nfs_default_protocol + 0}
-    default:   { fail('autofs::mount_nfs_default_protocol is not an integer.') }
-  }
-
-  case type3x($use_nis_maps) {
-    'string':  { $use_nis_maps_bool = str2bool($use_nis_maps) }
-    'boolean': { $use_nis_maps_bool = $use_nis_maps }
-    default:   { fail('autofs::use_nis_maps is not a boolean.') }
-  }
-
-  case type3x($maps_hiera_merge) {
-    'string':  { $maps_hiera_merge_bool = str2bool($maps_hiera_merge) }
-    'boolean': { $maps_hiera_merge_bool = $maps_hiera_merge }
-    default:   { fail('autofs::maps_hiera_merge is not a boolean.') }
-  }
-
-  case type3x($use_dash_hosts_for_net) {
-    'string':  { $use_dash_hosts_for_net_bool = str2bool($use_dash_hosts_for_net) }
-    'boolean': { $use_dash_hosts_for_net_bool = $use_dash_hosts_for_net }
-    default:   { fail('autofs::use_dash_hosts_for_net is not a boolean.') }
-  }
-
-  case type3x($service_enable) {
-    'string':  { $service_enable_bool = str2bool($service_enable) }
-    'boolean': { $service_enable_bool = $service_enable }
-    default:   { fail('autofs::service_enable is not a boolean.') }
-  }
-
-  case $maps_hiera_merge_bool {
+  case $maps_hiera_merge {
```

Comment:
Should we opt to use native Hiera merge in this module as well or should we keep the functionality in mode?

Reply:
You are right, I have removed hiera merge logic with a later commit.